lib/api/multiObjectDelete.js

```diff
@@ -412,11 +418,21 @@ function multiObjectDelete(authInfo, request, log, callback) {
                             log.error('wrong arn format from vault');
                             return next(errors.InternalError);
                         }
+                        const key = result.arn.slice(slashIndex + 1);
+                        const versionId = result.versionId ?
+                            versionIdUtils.encode(result.versionId) : undefined;
                         if (result.isAllowed) {
-                            inPlay.push(objects[i]);
+                            inPlay.push({
+                                key,
+                                versionId,
+                                decodedVid: result.versionId,
```

Comment:
I think this is wrong. We should be using the encoded version id sent by user for policies and so result.versionId should be the encoded version id. I need to look at this again